src/production/class.rs

```diff
@@ -86,3 +99,12 @@ impl<T: Class + core::fmt::Debug> core::fmt::Debug for ClassAllocation<T> {
         write!(f, "{:?}", inner)
     }
 }
+
+impl<T: Class> Clone for ClassAllocation<T> {
+    fn clone(&self) -> Self {
+        unsafe {
+            self.inner.add_ref();
+            core::mem::transmute_copy(self)
```

Comment:
Having to use `transmute` here makes me sad, but I get why we need to.

Reply:
Agreed.  Couldn't think of a better way to do it.